### PR TITLE
Add AWS config controls for Well-Architected Security Pillar & Reliability Pillar framework closes #588

### LINKED ISSUE #588: Add AWS config controls for Well-Architected Security Pillar & Reliability Pillar framework
**Is your feature request related to a problem? Please describe.**

The following controls are to be added as new config controls to address in AWS [WellArchitected](https://github.com/turbot/steampipe-mod-aws-well-architected)

- [ ] apigateway_rest_api_stage_xray_tracing_enabled
- [ ] ecs_task_definition_container_non_privileged
- [ ] ecs_task_definition_container_readonly_root_filesystem
- [ ] ecs_cluster_container_insights_enabled
- [ ] ecs_service_fargate_using_latest_platform_version
- [ ] efs_access_point_enforce_user_identity
- [ ] elb_classic_lb_multiple_az_configured
- [ ] lambda_function_use_latest_runtime
- [ ] lambda_function_multiple_az_configured
- [ ] opensearch_domain_in_vpc
- [ ] opensearch_domain_encryption_at_rest_enabled
- [ ] vpc_network_acl_remote_administration
- [ ] networkfirewall_firewall_policy_default_stateless_action_check_fragmented_packets
- [ ] networkfirewall_firewall_policy_default_stateless_action_check_full_packets
- [ ] networkfirewall_firewall_policy_rule_group_not_empty
- [ ] networkfirewall_stateless_rule_group_not_empty
- [ ] codebuild_project_s3_logs_encryption_enabled
- [ ] elb_application_gateway_network_lb_multiple_az_configured
- [ ] ec2_instance_not_use_multiple_enis

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by madhushreeray30**
New controls added :

- [x] apigateway_rest_api_stage_xray_tracing_enabled
- [x] ecs_task_definition_container_non_privileged
- [x]  ecs_task_definition_container_readonly_root_filesystem
- [x]  ecs_cluster_container_insights_enabled
- [x]  ecs_service_fargate_using_latest_platform_version
- [x]  efs_access_point_enforce_user_identity
- [x]  elb_classic_lb_multiple_az_configured
- [x]  lambda_function_use_latest_runtime
- [x]  lambda_function_multiple_az_configured
- [x]  opensearch_domain_in_vpc
- [x]  opensearch_domain_encryption_at_rest_enabled
- [x]  vpc_network_acl_remote_administration
- [x]  networkfirewall_firewall_policy_default_stateless_action_check_fragmented_packets
- [x]  networkfirewall_firewall_policy_default_stateless_action_check_full_packets
- [x]  networkfirewall_firewall_policy_rule_group_not_empty
- [x]  networkfirewall_stateless_rule_group_not_empty
- [x]  codebuild_project_s3_logs_encryption_enabled
- [x]  elb_application_gateway_network_lb_multiple_az_configured
- [x]  ec2_instance_not_use_multiple_enis

New queries + controls added :
- [x]  vpc_peering_dns_resolution_check
- [x] opensearch_domain_node_to_node_encryption_enabled
- [x] opensearch_domain_data_node_fault_tolerance
- [x] elb_network_lb_cross_zone_load_balancing_enabled
- [x] ecs_task_definition_nonroot_user

### CODE REVIEW

File: conformance_pack/networkfirewall.sp [hunk 1]
```diff
@@ -1,4 +1,38 @@
-# Non-Config rule query
+control "networkfirewall_firewall_policy_default_stateless_action_check_fragmented_packets" {
+  title       = "The default stateless action for Network Firewall policies should be drop or forward for fragmented packets"
```
Comment (khushboo9024) [inline conformance_pack/networkfirewall.sp:2]:
```suggestion
  title       = "Network Firewall policies default stateless action for fragmented packets should be drop or forward for fragmented packets"
```

File: conformance_pack/networkfirewall.sp [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+  description = "This control checks whether the default stateless action for fragmented packets for a Network Firewall policy is drop or forward. The control passes if Drop or Forward is selected, and fails if Pass is selected."
+  query       = query.networkfirewall_firewall_policy_default_stateless_action_check_fragmented_packets
+
+  tags = merge(local.conformance_pack_vpc_common_tags, {
+  })
+}
+
+control "networkfirewall_firewall_policy_default_stateless_action_check_full_packets" {
+  title       = "The default stateless action for Network Firewall policies should be drop or forward for full packets"
```
Comment (khushboo9024) [inline conformance_pack/networkfirewall.sp:11]:
```suggestion
  title       = "Network Firewall policies default stateless action should be drop or forward for full packets"
```

File: conformance_pack/networkfirewall.sp [hunk 3]
```diff
@@ -1,4 +1,38 @@
...
+  title       = "Network Firewall policies should have at least one rule group associated"
+  description = "This control checks whether a Network Firewall policy has any stateful or stateless rule groups associated. The control fails if stateless or stateful rule groups are not assigned."
+  query       = query.networkfirewall_firewall_policy_rule_group_not_empty
+
+  tags = merge(local.conformance_pack_vpc_common_tags, {
+  })
+}
+
+control "networkfirewall_stateless_rule_group_not_empty" {
+  title       = "Stateless network firewall rule group should not be empty"
```
Comment (khushboo9024) [inline conformance_pack/networkfirewall.sp:35]:
```suggestion
  title       = "Network firewall policies should not have an empty stateless rule group"
```

File: conformance_pack/ecs.sp [hunk 4]
```diff
@@ -71,6 +71,51 @@ control "ecs_task_definition_logging_enabled" {
...
+  title       = "ECS fargate services should run on the latest fargate platform version"
+  description = "This control checks if Amazon ECS Fargate services are running the latest Fargate platform version. This control fails if the platform version is not the latest."
+  query       = query.ecs_service_fargate_using_latest_platform_version
+
+  tags = merge(local.conformance_pack_ecs_common_tags, {
+  })
+}
+
+control "ecs_task_definition_nonroot_user" {
+  title       = "ECS task definition should not use root user"
```
Comment (khushboo9024) [inline conformance_pack/ecs.sp:84]:
```suggestion
  title       = "ECS task definitions should not use root user"
```

File: conformance_pack/opensearch.sp [hunk 5]
```diff
@@ -1,43 +1,37 @@
-# Non-Config rule query
+control "opensearch_domain_in_vpc" {
+  title       = "Amazon OpenSearch domains should be in a VPC"
```
Comment (khushboo9024) [inline conformance_pack/opensearch.sp:2]:
```suggestion
  title       = "OpenSearch domains should be in a VPC"
```

File: conformance_pack/vpc.sp [hunk 6]
```diff
@@ -275,10 +275,27 @@ control "vpc_network_acl_unused" {
   })
 }
 
+control "vpc_network_acl_remote_administration" {
+  title       = "Ensure no Network ACLs allow ingress from 0.0.0.0/0 to remote server administration ports"
```
Comment (khushboo9024) [inline conformance_pack/vpc.sp:279]:
```suggestion
  title       = "Network ACLs should not allow ingress from 0.0.0.0/0 to port 22 or port 3389"
```

File: conformance_pack/ecs.sp [hunk 7]
```diff
@@ -232,7 +277,46 @@ query "ecs_service_load_balancer_attached" {
...
+query "ecs_task_definition_nonroot_user" {
+  sql = <<-EOQ
+  select
+    task_definition_arn as resource,
+    case
+      when c ->> 'User' is null or c ->> 'User' = 'root' then 'alarm'
+      else 'ok'
+    end as status,
+    case
+      when c ->> 'User' is null or c ->> 'User' = 'root' then title || ' has root user.'
```
Comment (khushboo9024) [inline conformance_pack/ecs.sp:310]:
```suggestion
      when c ->> 'User' is null or c ->> 'User' = 'root' then title || ' uses root user to run EC2 launch type containers.'
```

File: conformance_pack/ecs.sp [hunk 8]
```diff
@@ -232,7 +277,46 @@ query "ecs_service_load_balancer_attached" {
...
+      else title || ' has non-root user.'
```
Comment (khushboo9024) [inline conformance_pack/ecs.sp:311]:
```suggestion
      else title || ' uses non-root user to run EC2 launch type containers.'
```

**Review by khushboo9024 — APPROVED** [review]

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!!

**Comment by khushboo9024** [comment]
Not required as these controls are defined in `Well-Architected` mod itself.

**Pull request closed without merging**